### PR TITLE
Check for autoload cookies on private functions

### PR DESCRIPTION
**Pull request by alphapapa**

### CODE REVIEW

**Comment by purcell** [comment]
Great, thanks.

File: package-lint.el [hunk 1]
```diff
@@ -261,6 +262,21 @@ This is bound dynamically while the checks run.")
 
 ;;; Checks
 
+(defun package-lint--check-autoloads-on-private-functions ()
+  "Verify that private functions don't have autoload cookies."
+  (goto-char (point-min))
+  (while (re-search-forward (rx bol "(defun" (1+ space)
```
Comment (Fanael) [inline package-lint.el]: This really should use the definitions from `package-lint--get-defs`.

Comment (purcell) [inline package-lint.el]: Good catch, thanks for fixing this.

Comment (alphapapa) [inline package-lint.el]: Thanks.

**Pull request merged**